integration-tests/python/common/events.py

```diff
@@ -0,0 +1,65 @@
+import time
+from abc import ABC, abstractmethod
+from collections import defaultdict
+
+from boltdb import BoltDB
+
+from common.process import Process
+
+class Events(ABC):
+    @abstractmethod
+    def processes(self):
+        pass
+
+    @abstractmethod
+    def network(self):
+        pass
+
+class BoltDBEvents(Events):
+    def __init__(self):
+        # lazy load the DB on demand, to avoid
+        # weird behavior when opening the DB really 
+        # early in the test
+        self.db = None
+
+    def await_processes(self, container, timeout=30):
+        for _ in range(timeout):
+            with self.open().view() as tx:
+                process_bucket = tx.bucket(b'Process')
+                if process_bucket:
+                    container_bucket = process_bucket.bucket(container.encode())
+                    if container_bucket:
+                        return
+                    
+            time.sleep(1)
+
+        raise RuntimeError(f'Timed out waiting for processes for {container}')
+
+    def processes(self, container=None):
```

Comment:
Use a more descriptive function name. Also add a comment explaining what the function does.
```suggestion
    # Gets processes from BoltDB. If container is not set all processes will be returned.
    # If container is set, only processes for that container will be returned.
    def processes(self, container=None):
```